net-mail/cyrus-imapd/cyrus-imapd-3.6.7.ebuild

```diff
@@ -0,0 +1,277 @@
+
+	# Workaround runtime crash
+	# bug #834573
+	append-flags -fno-toplevel-reorder
+
+	# Uses a lot of function pointers with undeclared function arguments
+	append-cflags -std=gnu17
+
+	# lto-type-mismatch
+	filter-lto
+
+	# TODO:
+	# - revisit --with-sphinx-build=no? (it's docs this time, not the search engine)
+	# - post-emerge message re lmdb removal?
+	local myconf=(
+		--enable-murder
+		--disable-release-checks
+		--enable-idled
+		--enable-autocreate
+		--enable-pcre
+		--with-com_err
+		--with-cyrus-user=cyrus
+		--with-sasl
+		--with-sphinx-build=no
+		--without-krb
+		--without-krbdes
+		--enable-squat
+		--with-zlib
+		--without-wslay
+		--without-chardet
+		--without-cld2
+		--disable-srs
+		$(use_enable afs)
+		$(use_enable afs krb5afspts)
+		$(use_enable backup)
+		$(use_enable calalarm calalarmd)
+		$(use_with caps libcap)
+		$(use_with clamav)
+		$(use_enable nntp)
+		$(use_enable http)
+		$(use_with http nghttp2)
+		$(use_enable replication)
+		$(use_enable kerberos gssapi)
+		$(use_with ldap)
+		$(use_with mysql)
+		$(use_with postgres pgsql)
+		$(use_with perl)
+		# TODO: optional again? idled etc require it
+		--with-sqlite
+		--enable-sieve
+		$(use_with ssl openssl)
+		--enable-server # bug #924652
+		$(use_enable static-libs static)
+		$(use_with tcpd libwrap)
+		$(use_enable xapian)
+		$(use_enable test unit-tests)
+	)
+
+	if use afs ; then
+		myconf+=(
+			--with-afs-libdir=/usr/$(get_libdir)
+			--with-afs-incdir=/usr/include/afs
+		)
+	fi
+
+	# http implicitly requires jmap, jmap explicitly requires http and xapian
+	use http && myconf+=( --enable-jmap )
+
+	econf ${myconf[@]}
+}
+
+src_install() {
+	emake DESTDIR="${D}" INSTALLDIRS=vendor install
+
+	dodoc README*
+	dodoc -r doc
+
+	cp -r contrib tools "${ED}"/usr/share/doc/${PF} || die
+	rm -f doc/text/Makefile* || die
+
+	mv "${ED}"/usr/libexec/{master,cyrusmaster} || die
+
+	insinto /etc
+	newins "${ED}"/usr/share/doc/${PF}/doc/examples/cyrus_conf/normal.conf cyrus.conf
+	newins "${ED}"/usr/share/doc/${PF}/doc/examples/imapd_conf/normal.conf imapd.conf
+
+	sed -i -e '/^configdirectory/s|/var/.*|/var/imap|' \
+		-e '/^partition-default/s|/var/.*|/var/spool/imap|' \
+		-e '/^sievedir/s|/var/.*|/var/imap/sieve|' \
+		"${ED}"/etc/imapd.conf || die
+
+	sed -i -e 's|/var/imap/socket/lmtp|/run/cyrus/socket/lmtp|' \
+		-e 's|/var/imap/socket/notify|/run/cyrus/socket/notify|' \
+		"${ED}"/etc/cyrus.conf || die
+
+	# same thing for http(s) as well
+	if ! use http; then
+		sed -i -e "/http/s/^/#/" "${ED}"/etc/cyrus.conf || die
+	fi
+
+	newinitd "${FILESDIR}"/cyrus.rc8 cyrus
+	newconfd "${FILESDIR}"/cyrus.confd cyrus
+	newpamd "${FILESDIR}"/cyrus.pam-include sieve
+
+	for subdir in imap/{,db,log,msg,proc,socket,sieve} spool/imap/{,stage.} ; do
+		keepdir "/var/${subdir}"
+		fowners cyrus:mail "/var/${subdir}"
+		fperms 0750 "/var/${subdir}"
+	done
+
+	for subdir in imap/{user,quota,sieve} spool/imap ; do
+		for i in a b c d e f g h i j k l m n o p q r s t v u w x y z ; do
```

Comment:
I'm not sure this is correct to do anymore (has been part of the ebuilds from at least 2.1 era) especially with the new default (from 3.6) of defaulting to storing mailboxes by uuid rather than name.

I note that the mkimap script does not create these and the dohash script was removed in 3.1.x (dev branch). I suspect the operations of the dohash script were the basis for this a-z creation.